vyper/ast/nodes.py

```diff
@@ -1290,6 +1291,14 @@ def __init__(self, *args, **kwargs):
             elif call_name == "public":
                 # declaring a public variable
                 self.is_public = True
+                # handle the cases where a constant or immutable variable is public
+                if not len(self.annotation.args):
```

Comment:
```suggestion
                # do the same thing as `validate_call_args`
                # (can't be imported due to cyclic dependency)
                if len(self.annotation.args) != 1:
```

Reply:
can you remove the corresponding dead code in vyper/semantics/validation/module.py?